ticket-granting/protocol.go

```diff
@@ -2,51 +2,54 @@ package ticket_granting
 
 import (
 	"github.com/PretendoNetwork/nex-go"
+	"github.com/PretendoNetwork/nex-go/types"
 	_ "github.com/PretendoNetwork/nex-protocols-go"
 	ticket_granting "github.com/PretendoNetwork/nex-protocols-go/ticket-granting"
 
 	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
 )
 
-var commonTicketGrantingProtocol *CommonTicketGrantingProtocol
-
-type CommonTicketGrantingProtocol struct {
-	*ticket_granting.Protocol
-	server                   *nex.Server
-	secureStationURL         *nex.StationURL
-	buildName                string
-	allowInsecureLoginMethod bool
-}
-
-func (commonTicketGrantingProtocol *CommonTicketGrantingProtocol) SetSecureStationURL(stationURL *nex.StationURL) {
-	commonTicketGrantingProtocol.secureStationURL = stationURL
-}
-
-func (commonTicketGrantingProtocol *CommonTicketGrantingProtocol) SetBuildName(buildName string) {
-	commonTicketGrantingProtocol.buildName = buildName
+var commonProtocol *CommonProtocol
+
+type CommonProtocol struct {
+	protocol                   ticket_granting.Interface
+	SecureStationURL           *types.StationURL
+	SpecialProtocols           []*types.PrimitiveU8
+	StationURLSpecialProtocols *types.StationURL
+	BuildName                  *types.String
+	allowInsecureLoginMethod   bool
+	SessionKeyLength           int
+	AccountDetailsByPID        func(pid *types.PID) (*nex.Account, uint32)
+	AccountDetailsByUsername   func(username string) (*nex.Account, uint32)
+	SecureServerAccount        *nex.Account
```

Comment:
Maybe we could take these values from the PRUDP endpoint, once HPP checks are sorted out?

Reply:
I'm open to the idea if you can propose an API. What do you propose for getting it from the endpoint? This protocol is supposed to be used on the authentication endpoint, which uses the authentication account. If we pull that endpoints account it won't be the correct one, since it needs the secure server account

Reply:
You are right about the secure server account, we can keep that. but maybe we could at least use the `AccountDetails` functions from the endpoint?

```go
	sourceAccount, errorCode := endpoint.AccountDetailsByUsername(strUserName.Value)
	if errorCode != nil && errorCode.ResultCode != nex.ResultCodes.RendezVous.InvalidUsername {
		// * Some other error happened
		return nil, errorCode.ResultCode
	}

	targetAccount, errorCode := endpoint.AccountDetailsByUsername(commonProtocol.SecureServerAccount.Username)
	if errorCode != nil && errorCode.ResultCode != nex.ResultCodes.RendezVous.InvalidUsername {
		// * Some other error happened
		return nil, errorCode.ResultCode
	}
```

Reply:
Sounds good to me :+1: